distage/distage-core-api/src/main/scala/izumi/distage/model/definition/Lifecycle.scala

```diff
@@ -1114,4 +1114,76 @@ object Lifecycle {
 
   @deprecated("renamed to fromAutoCloseable", "0.11")
   def fromAutoCloseableF[F[_], A <: AutoCloseable](acquire: => F[A])(implicit F: DIEffect[F]): Lifecycle[F, A] = fromAutoCloseable(acquire)
+
+
+  /**
+    * This type encoding was copy-pasted from cats.effect.Resource
+    * It's needed to declare a new type with cats.Applicative
+    * capabilities of doing parallel processing in `ap` and `map2`, needed
+    * for implementing `cats.Parallel`.
+    */
+  type Par[+F[_], +A] = Par.Type[F, A]
```

Comment:
This isn't actually necessary, you may just declare
```
type F[x] =  Lifecycle[F0, x]
```
In instance and it won't be visible outside that `F=Lifecycle` because you do not specify what is `type F` in the output already `: Parallel[Lifecycle[F0, ?]]`